crates/trigger/src/stdio.rs

```diff
@@ -56,13 +56,19 @@ impl StdioLoggingTriggerHooks {
         &self,
         component_id: &str,
         log_suffix: &str,
-        log_dir: &Path,
+        log_dir: Option<&Path>,
     ) -> Result<ComponentStdioWriter> {
         let sanitized_component_id = sanitize_filename::sanitize(component_id);
-        let log_path = log_dir.join(format!("{sanitized_component_id}_{log_suffix}.txt"));
+        let log_path = log_dir
+            .map(|log_dir| log_dir.join(format!("{sanitized_component_id}_{log_suffix}.txt",)));
+        let log_path = log_path.as_deref();
```

Comment:
Can this be tacked onto the end of the `map` (`let log_path = log_dir.map(...).as_deref()`) or do you need the intermediate value to satisfy the savage borrow checker?

Reply:
Not the borrow checker per se; you can't return a reference to an owned value (the result of `join`) from a closure.

The borrow on this line is only allowed because of the special "temporary lifetime extension" rule which applies to the local scope of a `let`.